Jenkinsfile

```diff
@@ -19,11 +19,20 @@ def mavenEnv(Map params = [:], Closure body) {
       timeout(120) {
         withChecks(name: 'Tests', includeStage: true) {
           infra.withArtifactCachingProxy {
-            withEnv([
-              'JAVA_HOME=/opt/jdk-' + params['jdk'],
-              "MAVEN_ARGS=${env.MAVEN_ARGS != null ? MAVEN_ARGS : ''} -B -ntp -Dmaven.repo.local=${WORKSPACE_TMP}/m2repo"
+            cache(
+                // max cache size in MB, will be reset after exceeding this size
+                maxCacheSize: 2048
+                defaultBranch: 'master', caches: [
+                      arbitraryFileCache(path: "${WORKSPACE_TMP}/m2repo",
+                      cacheValidityDecidingFile: '.repository-cache-marker'
```

Comment:
we can't use a `pom.xml` as it will be updated on every PR and never used

Reply:
I understood it is the `pom.xml` per dependency (e.g. in `~/.m2/repository/**/*`): did I misunderstood?

Reply:
this is:

> The workspace-relative path to one or multiple files which should be used to determine whether the cache is up-to-date or not. Only up-to-date caches will be restored and only outdated caches will be created.

if the files content changes the whole cache will be invalided, i.e. the cache would be invalided multiple times per day and be pretty worthless...

Reply:
correct, I don't think we want that